src/test/ui/or-patterns/or-patterns-syntactic-fail.stderr

```diff
@@ -0,0 +1,112 @@
+error: expected one of `:` or `@`, found `|`
+  --> $DIR/or-patterns-syntactic-fail.rs:28:14
+   |
+LL |     fn fun(A | B: E) {}
+   |              ^ expected one of `:` or `@` here
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:37:11
+   |
+LL |     let ( | A | B) = E::A;
+   |           ^ help: remove the `|`
```

Comment:
Could we move these tests to its own test file with `run-rustfix`?

Reply:
Sure; I would prefer duplicating them to another test file tho rather than moving them since this file is supposed to act as a specification test.